L/LibCURL/common.jl

```diff
@@ -0,0 +1,111 @@
+        --without-spnego --without-libpsl --disable-ares --disable-manual
+        --disable-ldap --disable-ldaps --without-zsh-functions-dir
+        --disable-static --without-libgsasl
+
+        # A few things we actually enable
+        --with-libssh2=${prefix} --with-zlib=${prefix} --with-nghttp2=${prefix}
+        --enable-versioned-symbols
+    )
+
+
+    if [[ ${target} == *mingw* ]]; then
+        # We need to tell it where to find libssh2 on windows
+        FLAGS+=(LDFLAGS="${LDFLAGS} -L${prefix}/bin")
+
+        # We also need to tell it to link against schannel (native TLS library)
+        FLAGS+=(--with-schannel)
+    elif [[ ${target} == *darwin* ]]; then
+        # On Darwin, we need to use SecureTransport (native TLS library)
+        FLAGS+=(--with-secure-transport)
+
+        # We need to explicitly request a higher `-mmacosx-version-min` here, so that it doesn't
+        # complain about: `Symbol not found: ___isOSVersionAtLeast`
+        if [[ "${target}" == *x86_64* ]]; then
+            export CFLAGS=-mmacosx-version-min=10.11
+        fi
+    else
+        # On all other systems, we use MbedTLS
+        FLAGS+=(--with-mbedtls=${prefix})
+    fi
+
+    ./configure --prefix=$prefix --host=$target --build=${MACHTYPE} "${FLAGS[@]}"
+    make -j${nproc}
+    if [[ "${THIS_IS_CURL}" == true ]]; then
+        # Manually install only `curl`
+        install -Dm 755 "src/.libs/curl${exeext}" "${bindir}/curl${exeext}"
+    else
+        # Install everything...
+        make install
+        # ...but remove `curl`
+        rm "${bindir}/curl${exeext}"
+    fi
+    install_license COPYING
+    """
+
+    # These are the platforms we will build for by default, unless further
+    # platforms are passed in on the command line
+    platforms = supported_platforms()
+
+    # The products that we will ensure are always built
+    if this_is_curl_jll
+        # CURL_jll only provides the executable
+        products = [
+            ExecutableProduct("curl", :curl),
+        ]
+    else
+        # LibCURL only provides the library
+        products = [
+            LibraryProduct("libcurl", :libcurl),
+        ]
+    end
+
+    # Dependencies that must be installed before this package can be built
+    dependencies = [
+        Dependency("LibSSH2_jll"),
+        Dependency("Zlib_jll"),
+        Dependency("nghttp2_jll"),
+        # Note that while we unconditionally list MbedTLS as a dependency,
+        # we default to schannel/SecureTransport on Windows/MacOS.
+        Dependency("MbedTLS_jll"; compat="~2.28.0", platforms=filter(p->Sys.islinux(p) || Sys.isfreebsd(p), platforms)),
+    ]
+
+    if this_is_curl_jll
+        # Curl_jll depends on LibCURL_jll
+        push!(dependencies, Dependency("LibCURL_jll"; compat="$(version)"))
+    end
+
+    # Build the tarballs, and possibly a `build.jl` as well.
+    build_tarballs(ARGS, name, version, sources, script, platforms, products, dependencies; julia_compat="1.8")
```

Comment:
Why 1.8 only?

Reply:
https://github.com/JuliaLang/julia/blob/release-1.7/stdlib/LibCURL_jll/Project.toml#L3
https://github.com/JuliaLang/julia/blob/release-1.8/stdlib/LibCURL_jll/Project.toml#L3

Reply:
Oh… I guess that’s why we’d want a fully static thing like you mentioned. Annoying. Thanks for your help with this!

Reply:
@giordano @staticfloat What if we:

1. Don't make CURL_jll depend on LibCURL_jll
2. When building CURL_jll, don't delete `libcurl`. So then CURL_jll will supply both `libcurl` and `curl`.

Would that allow us to use the same version of `CURL_jll` on multiple versions of Julia?

Reply:
https://github.com/JuliaPackaging/Yggdrasil/pull/4472#issuecomment-1046852810

Reply:
I see, so we'd need static libraries for all of the dependencies.

Reply:
So this just means that we need to remember to make a new release of CURL_jll every time there's a new minor release of Julia, right?

Reply:
No, just always build `CURL_jll` together with (well, right after) a new LibCURL_jll.

Reply:
> No, just always build `CURL_jll` together with (well, right after) a new LibCURL_jll.

It would be nice if we could automate that process.

Reply:
https://github.com/JuliaPackaging/BinaryBuilder.jl/issues/778.  This is what https://github.com/JuliaPackaging/Yggdrasil/pull/4472#issuecomment-1050346049 and https://github.com/JuliaPackaging/Yggdrasil/blob/2fa71a47eedfe3f500896f5aa8bf20306e8049ae/H/HarfBuzz/common.jl#L17-L19 were referring to.